### PR TITLE
issue-11: Implemented countEven method and created tests for it.

### LINKED ISSUE #16: issue-11: method countEven
1. Add a new method to UtilityMethods for finding the number of even elements in the array.
2. the method should have the signature: ``public static int countEven(int[] array)``

Tips for implementation:

1. check that the input array is not null using ``com.google.common.base.Preconditions.checkNotNull`` method
3. use existing `isEven` method to check if an element is even
2. add unit test

### PR DESCRIPTION
**Pull request by ponomvrenko**

### CODE REVIEW

File: src/main/java/org/example/utility/UtilityMethods.java [hunk 1]
```diff
@@ -24,4 +26,13 @@ public static boolean isEven(int a) {
     public static boolean isOdd(int a) {
         return !isEven(a);
     }
+
+    public static int countEven(int[] array) {
+        Preconditions.checkNotNull(array, "Your array is null!");
+        int evenElements = 0;
+        for (int e : array) {
+            if (isEven(e)) evenElements++;
```
Comment (kojote-git) [inline src/main/java/org/example/utility/UtilityMethods.java]: We should use curly braces with if, for and other statements. As an example see "[Google Java Style Guide: 4.1.1 Braces are used where optional](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)".

FYI: see also the reason why we should use curly braces: https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces

So, please,  use curly braces :)

File: src/main/java/org/example/utility/UtilityMethods.java [hunk 2]
```diff
@@ -24,4 +26,13 @@ public static boolean isEven(int a) {
     public static boolean isOdd(int a) {
         return !isEven(a);
     }
+
+    public static int countEven(int[] array) {
+        Preconditions.checkNotNull(array, "Your array is null!");
```
Comment (kojote-git) [inline src/main/java/org/example/utility/UtilityMethods.java]: It's not a strict requirement, but I think it's better to have a static import of the checkNotNull method here. One reason is that other developers from our team have already used the static import in their pull requests. So I think we should have a consistency here

File: src/test/java/org/exmple/utility/UtilityMethodsTest.java [hunk 3]
```diff
@@ -3,9 +3,7 @@
 import org.example.utility.UtilityMethods;
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (kojote-git) [inline src/test/java/org/exmple/utility/UtilityMethodsTest.java]: We should not have wildcard imports. As an example, see "[Google Java Style Guide: 3.3.1 No wildcard imports](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports)"
We should use explicit imports instead.

FYI: "[Why is using a wild card with a Java import statement is bad](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad)"

File: src/test/java/org/exmple/utility/UtilityMethodsTest.java [hunk 4]
```diff
@@ -74,4 +72,25 @@ public void isOdd_shouldReturnFalse() {
 
         assertFalse(result);
     }
+
+    @Test
+    public void countEven_shouldReturnTrue() {
```
Comment (kojote-git) [inline src/test/java/org/exmple/utility/UtilityMethodsTest.java]:
```suggestion
    public void countEven_shouldCountEvenNumers() {
```

File: src/test/java/org/exmple/utility/UtilityMethodsTest.java [hunk 5]
```diff
@@ -74,4 +72,25 @@ public void isOdd_shouldReturnFalse() {
...
+        int[] array = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+        int result = UtilityMethods.countEven(array);
+
+        assertEquals(5, result);
+    }
+
+    @Test
+    public void countEven_shouldReturnNullPointerException() {
```
Comment (kojote-git) [inline src/test/java/org/exmple/utility/UtilityMethodsTest.java]:
```suggestion
    public void countEven_shouldThrowNullPointerException() {
```

**Comment by ponomvrenko** [comment]
I've taken into account all the comments and have corrected everything in this commit.

**Pull request merged**